packages/abcm2ps/PKGBUILD

```diff
@@ -0,0 +1,33 @@
+# Maintainer: OSAMC <https://github.com/osam-cologne/archlinux-proaudio>
+# Contributor: Christopher Arndt <aur -at- chrisarndt -dot- de>
+# Contributor: Tuyen Pham <tuyenpm9@gmail.com>
+# Contributor: Lex Black <autumn-wind@web.de>
+# Contributor: Simon Thorpe <simon@hivetechnology.com.au>
+# Contributor: kess <joe@macmahon.eu>
+# Contributor: damir <damir@archlinux.org>
+# Contributor: Luis Pablo Gasparotto <lpgasparotto(at)gmail(dot)com>
+
+pkgname=abcm2ps
+pkgver=8.14.13
+pkgrel=2
+pkgdesc='Convert ABC music notation files to PostScript from the commandline'
+arch=(x86_64 aarch64)
+url='http://moinejf.free.fr/'
+license=(GPL3)
+depends=(pango)
```

Comment:
`abcm2ps` links these libraries:

    -lpangocairo-1.0 -lcairo -lpangoft2-1.0 -lpango-1.0 -lgobject-2.0 -lglib-2.0 -lharfbuzz -lfontconfig -lfreetype -lm

But these are all indirect dependencies of `pango'. I'm still a bit confused about the "declare *transient dependencies*" rule. How do I detect, whether `abcm2ps` needs e.g. `glib2` directly, or whether it is only needed be one (or several)  of the other libraries? Or asked another way, if pango stopped depending (indirectly) on `glib2` and thus wouldn't install it, would `abcm2ps` still work?

Reply:
The [configure script uses pkg-config](https://github.com/lewdlime/abcm2ps/blob/master/configure#L66):
```
CPPFLAGS="$CPPFLAGS -DHAVE_PANGO=1 `$PKG_CONFIG pango cairo freetype2 --cflags`"
LDLIBS="`$PKG_CONFIG pangocairo pangoft2 freetype2 --libs` $LDLIBS"
```
Based on this I'd just list pango, cairo and freetype as dependencies.

Reply:
If pango stopped depending on glib2 and abcm2ps actually needed the glib2 itself our containerized build system will give an error, right? At least that is what I remember from the planning talks with @dvzrv

Reply:
@diovudau This is about run-time, i.e. if the user (hypothetically) installed a pang package, which doesn't require glib2 anymore and glib2 was marked as only installed as a dependency and thus would get de-installed, would abcm2ps still work

TBH, I find this "transitive dependencies" thing highly theoretical anyway. IMHO, in practice this changing of indirect dependencies rarely happens, and, as you say, it will be usually could at build time. Things were easier when the rule was "only include dependencies not already covered by other dependencies", *grumble*.

Reply:
FWIW: You can use [find-libdeps](https://man.archlinux.org/man/find-libdeps.1) to identify all shared objects directly linked against.

Reply:
> Based on this I'd just list pango, cairo and freetype as dependencies.

Well, you see, based on the output of 

> FWIW: You can use [find-libdeps](https://man.archlinux.org/man/find-libdeps.1) to identify all shared objects directly linked against.

```
libpangocairo-1.0.so=0-64
libpangoft2-1.0.so=0-64
libpango-1.0.so=0-64
libgobject-2.0.so=0-64
libglib-2.0.so=0-64
libfreetype.so=6-64
libm.so=6-64
libc.so=6-64
```

`cairo` is not among the packages which provides any of the these libraries. Do you see why I find this confusing? ;)

Based on the above, I would now arrive at:

```
depends=(glibc)
makedepends=(freetype2 glib2 pango python-docutils)
```

and in `package()`:

```
  depends+=(libpangocairo-1.0.so libpangoft2-1.0.so libpango-1.0.so
            libgobject-2.0.so libglib-2.0.so libfreetype.so)
```

and `cairo` would be supplied by the make dependency on `pango`. The direct mentioning of `cairo` in abcm2ps' Makefile is a product of autotools.

Anybody have another interpretation?